### PR TITLE
Fixes grid origins in TerrainImporter to match Isaac Sim cloner

### LINKED ISSUE #287: [Proposal] Inconsistent Grid Cloning
### Proposal

The logic for grid cloning in grid_cloner.py (func: `get_clone_transforms()`) and in terrain_importer.py (func: `_compute_env_origins_grid()`) are different, consequently they give inconsistent values. Is there any particular reasoning/need for it?

### Motivation

While initializing a new entity, I'd like to save some values from the environments during its initialization. However, during the initialization of an entity, the grid locations are based on the `env_origins` generated by grid_cloner.py but later overridden by the one in terrain_importer.py. This feature can be beneficial in avoiding recomputation of values that do not change in an environment, hence speeding up the training. In my case, these values were the poses of sampled points on a static object in the world frame. These values can be stored during initialization, and the distances from the robot base to these points could be updated as required.

### Alternatives

I'm currently using a quadruped on a flat ground interacting with a static object. Hence, updating `_compute_env_origins_grid()` as follows would suffice to make the logic match:

```python
def _compute_env_origins_grid(self, num_envs: int, env_spacing: float) -> torch.Tensor:
    """Compute the origins of the environments in a grid based on configured spacing."""
  
    num_rows = int(np.ceil(np.sqrt(num_envs)))  # Calculate number of rows based on total environments
    num_cols = int(np.ceil(num_envs / num_rows))  # Adjust number of columns based on rows
  
    env_origins = torch.zeros(num_envs, 3, device=self.device)
  
    for i in range(num_envs):
        row = i // num_cols
        col = i % num_cols
  
        # Calculate position with row-first approach
        x = -(row - (num_rows - 1) / 2) * env_spacing
        y = (col - (num_cols - 1) / 2) * env_spacing
  
        env_origins[i, 0] = x
        env_origins[i, 1] = y
        env_origins[i, 2] = 0.0
  
    return env_origins
```

Another possible solution would be to ensure that all entities are added to the scene only after the final grid configuration is computed, i.e., in this case, after `_compute_env_origins_grid()` is called. This would be a more permanent solution.
### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

### Acceptance Criteria

- [x] Update logic of `_compute_env_origins_grid()` in terrain_importer.py
- [ ] Add entities to the scene only after the final grid configuration is computed

### PR DESCRIPTION
**Pull request by shafeef901**
# Description
The logic for grid cloning in grid_cloner.py (func: `get_clone_transforms()`) and in terrain_importer.py (func: `_compute_env_origins_grid()`) are different, consequently they give inconsistent values. This PR fixes it by updating the logic of  `_compute_env_origins_grid()` to make it consistent with IsaacSim.

Fixes #287 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./orbit.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have run all the tests with `./orbit.sh --test` and they pass
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Comment by Mayankm96** [comment]
Can you please also update the changelog and version in extension.toml of the affected extensions?

**Comment by shafeef901** [comment]
Hi @Mayankm96 ,
I have added a unit test that compares the env_origins from both sources and then runs the sim for a couple of time steps.
Let me know if it looks good to be merged :)

**Review by Mayankm96 — APPROVED** [review]

**Comment by Mayankm96** [comment]
Looks good. Thanks a lot! :)

**Comment by Mayankm96** [comment]
Ah @shafeef901 can you please also update the extension.toml file with the version update? Thanks a lot for the test and the fix!

**Comment by shafeef901** [comment]
Done!

File: source/extensions/omni.isaac.orbit/omni/isaac/orbit/terrains/terrain_importer.py [hunk 1]
```diff
@@ -347,14 +347,16 @@ def _compute_env_origins_curriculum(self, num_envs: int, origins: torch.Tensor)
         return env_origins
 
     def _compute_env_origins_grid(self, num_envs: int, env_spacing: float) -> torch.Tensor:
-        """Compute the origins of the environments in a grid based on configured spacing."""
+        """Compute the origins of the environments in a grid based on configured spacing"""
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.orbit/omni/isaac/orbit/terrains/terrain_importer.py]:
```suggestion
        """Compute the origins of the environments in a grid based on configured spacing."""
```

File: source/extensions/omni.isaac.orbit/docs/CHANGELOG.rst [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+
+
 0.15.4 (2024-03-22)
 ~~~~~~~~~~~~~~~~~~~
 
 Fixed
 ^^^^^
 
-* Fixed the NonHolonomicActionCfg variable naming from joint_vel to _joint_vel_command to match the initialized variable in the init() function.
+* Fixed the NonHolonomicActionCfg variable22 naming from joint_vel to _joint_vel_command to match the initialized variable in the init() function.
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.orbit/docs/CHANGELOG.rst]:
```suggestion
* Fixed the NonHolonomicActionCfg variable naming from joint_vel to _joint_vel_command to match the initialized variable in the init() function.
```

**Review by Mayankm96 — APPROVED** [review]

**Comment by Mayankm96** [comment]
Thanks a lot for the MR. Merged with some small fixes to the test.

Mainly, instead of `assertTrue(torch.allclose)`, you can simply do [`torch.testing.all_close`](https://pytorch.org/docs/stable/testing.html). Also since you used the sim context through context manager, you don't need to write setUp and tearDown anymore :)

**Pull request merged**